cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/internal/replication/ReplicatorIdTest.java

```diff
@@ -14,33 +14,50 @@
 
 package com.cloudant.sync.internal.replication;
 
-import static org.mockito.Mockito.mock;
-
 import com.cloudant.common.CollectionFactory;
-import com.cloudant.sync.internal.documentstore.DatabaseImpl;
+import com.cloudant.sync.documentstore.DocumentStore;
 import com.cloudant.sync.replication.PullFilter;
 import com.cloudant.sync.replication.ReplicatorBuilder;
+import com.cloudant.sync.util.TestUtils;
 
+import org.junit.After;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 
+import java.io.File;
 import java.net.URI;
 
 /**
  * Created by tomblench on 10/12/14.
  */
 public class ReplicatorIdTest {
 
+    String testDir;
+    DocumentStore documentStore;
+
+    @Before
+    public void before() throws Exception {
+        testDir = TestUtils.createTempTestingDir(this.getClass().getName());
+        documentStore = DocumentStore.getInstance(new File(testDir));
+    }
+
+    @After
+    public void after() throws Exception {
+        documentStore.close();
+        TestUtils.deleteTempTestingDir(testDir);
+    }
+
     // source/target switched, so ids should be different
     @Test
     public void pullNotEqualToPush() throws Exception {
         PullStrategy pull = (PullStrategy) ((ReplicatorImpl) ReplicatorBuilder.pull()
                 .from(new URI
-                ("http://default-host/default-database")).to(mock(DatabaseImpl.class)).build
+                ("http://default-host/default-database")).to(documentStore).build
```

Comment:
@ricellis it's not currently possible to mock our DocumentStore implementation for the purposes of this test. In `PushStrategy` we can't mock `source.database` and in `PullStrategy` we can't mock `target.database` because they are final fields. So this is definitely an argument for turning `DocumentStore.database` and `DocumentStore.query` into getter methods.